docs/arch/adr-001-send-unpublish-emails-for-single-pages.md

```diff
@@ -0,0 +1,61 @@
+# Decision record: reinstate unpublish emails for single page subscriptions
+
+Date: 2021-12-03
+
+## Context
+
+We have just launched a new email notifications feature that allows users to subscribe to updates on a single page.
+Currently this feature is only enabled on a few selected pages (eg. [Open standards for government][open-standards-for-government]),
+but it is expected to roll out to the majority of pages and content types by early 2022.
+There is a user need to let subscribers know when a page they are subscribed to has been withdrawn or redirected.
+
+The email alert API used to send emails when a page in a taxon subscription was unpublished.
+This feature was [removed in Jan 2021][remove-previous-unpublish-emails] for good reasons.
+
+[open-standards-for-government]: https://www.gov.uk/government/publications/open-standards-for-government
+[remove-previous-unpublish-emails]: https://github.com/alphagov/email-alert-api/issues/1572
+
+## Decision
+
+We will reinstate emails when a page is unpublished, but only to users with a single page subscription to that page.
+We will also remove the subscriber list for a single page when it is unpublished to reduce the build up of empty lists in Email Alert API.
+
+### Addressing the concerns from the previous version
+
+The previous version of this feature was removed for several good reasons. We believe readding the unpublish emails only for single page subscriptions addresses these concerns:
+
+#### It only worked for topic taxons
+
+The new unpublished emails will only apply to single page subscriptions. This is a deliberate decision to keep the scope small and allow us to address the other concerns about handling an unpublished page in a taxon.
+
+#### It only worked if the unpublished page was redirected
+
+The new unpublished emails will handle pages that are redirected published in error with or without an alternative URL. This covers the publisher-facing [unpublishing types][unpublishing-types].
```

Comment:
It's probably worth noting that it would still be possible for an email to not go out but page ended - this could happen if the page was replaced with a redirect or a gone, rather than unpublished to be one. Unlikely but a potential edge case that'll occur for a wide rollout.

Reply:
Do you mean that there are two possible cases:

- An `unpublishing` message gets put on rabbitmq, which makes the page a `redirect` / `gone` / `vanish`
- The page is just directly replaced with a `redirect` / `gone` / `vanish` (which would be a message on the publishing queue) with no `unpublishing` first?

Reply:
@barrucadu we covered the `reidrect` and `gone` case, We mightw want to check in on taht `vanish` case. Not sure that is covered currently 🤔

Reply:
Given what we saw yesterday, I'm not sure we'll get a `vanish` as a result of an unpublishing - so we can probably ignore that case.

Also, I think we could leave the larger problem of handling a page being replaced without being unpublished as future work (but should call it out as such in the ADR).  We'd have to handle that by searching for a subscriberlist with the right `url`, as we use the content ID in the rules...